### PR TITLE
Rspect ris/endnote/refworks

### PR DESCRIPTION
**Pull request by MaggieZhaoYale**
To Fix issue #2 
- git fetch origin pull/15/head:citation_test
- git checkout citation_test
- bundle exec rspec spec/summon
Expect 19 examples, 0 failures.

### CODE REVIEW

File: lib/summon/ris.rb [hunk 1]
```diff
@@ -33,7 +33,7 @@ def ris_tags
 
       def ris_normal(value)
         if value.kind_of?(Array)
-          value.tag_per_value? ? value : [value.join(', ')]
+          value.tag_per_value? ? value : [value.join('; ')]
```
Comment (josh-levinson) [inline lib/summon/ris.rb:36]: I'm curious what's going on here. This looks like a change to the way the RIS library actually works?

File: lib/summon/ris_mappings.rb [hunk 2]
```diff
@@ -43,7 +43,6 @@
   'SN  -': ->() { issns.empty? ? isbns : issns }, #ISSN/ISBN
   'SP  -': ->() { start_page }, #Start Page
   'T1  -': ->() { subtitle ? "#{title}: #{subtitle}" : title}, #Primary Title
-  'T1  -': ->() { subtitle ? "#{title}: #{subtitle}" : title}, #Primary Title
```
Comment (josh-levinson) [inline lib/summon/ris_mappings.rb:46]: This looks fine (a duplicate line) but also is another line of actual code change in PR labeled as only a test change

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,2 +1,5 @@
 require 'summon/refworks'
-require 'summon/transport/canned'
\ No newline at end of file
+require 'summon/ris'
+require 'summon/endnote'
+require 'summon/transport/canned'
+require 'byebug'
```
Comment (josh-levinson) [inline spec/spec_helper.rb:5]: I don't know that we want to leave `byebug` in here, is it really required for testing or only when we need it?

File: spec/summon/ris_spec.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+describe Summon::Ris do
+  before do
+    @search = Summon::Service.new(:transport => Summon::Transport::Canned.new).search
+    @doc = @search.documents.first
+    #byebug
```
Comment (josh-levinson) [inline spec/summon/ris_spec.rb:7]: Is this something we want to leave in here?

File: summon-citation_export.gemspec [hunk 5]
```diff
@@ -19,4 +19,5 @@ Gem::Specification.new do |gem|
 
   gem.add_development_dependency 'rake'
   gem.add_development_dependency 'rspec'
+  gem.add_development_dependency 'byebug'
```
Comment (josh-levinson) [inline summon-citation_export.gemspec:22]: I'm not sure if this is something we want to do or not. Maybe have a discussion about it? Is this normal?

**Review by josh-levinson — CHANGES_REQUESTED** [review]
The PR works as intended and all of the tests pass as expected. I left a little feedback as follows:

1. It looks like there are 1-2 changes in the core library code outside of rspec and no reference to what is happening there or why?
2. `byebug` being added to the Gem as a requirement -- is this desired or expected? Is this normal in gem development?

Other than that it looks good.

File: lib/summon/ris.rb (hunk 1 above)
Comment (MaggieZhaoYale) [reply lib/summon/ris.rb:36]: The join dividend in ris was changed from ,  to `;` cause `,` is the same as `,` between last  name and first name.

Comment (josh-levinson) [reply lib/summon/ris.rb:36]: I guess what I'm wondering is if we are making changes to the library itself, if there are any additional tests we need to do to verify that the changes work/are safe?

Comment (MaggieZhaoYale) [reply lib/summon/ris.rb:36]: Wait, I was wrong. The change was already in test branch on June 10. Not sure why it shows the difference. https://github.com/yalelibrary/summon-citation_export/blob/test/lib/summon/ris.rb
There is a test to verify the change https://github.com/yalelibrary/summon-citation_export/blob/rpsect_ris/spec/summon/ris_spec.rb#L38

**Pull request closed without merging**